### DISCUSSION TITLE
Manual Server Installation

### DISCUSSION THREAD

**Discussion by joellti** [TriliumNext/Trilium#6007] [General]
I was unable to successfully complete the procedure at https://triliumnext.github.io/Docs/Wiki/manual-server-installation.html. On the other hand, I was able to complete the procedure for the original trilium notes at https://github.com/zadam/trilium/wiki/Manual-server-installation, which appears to be what this one is based on. I performed this procedure in both amd64 and arm64 platforms with the same result. Basically command `npm install` fails to complete without error. Maybe the procedure needs an update?

**Comment by CobriMediaJulien** [comment]
Why not using the reccomend way with docker?

**Reply by joellti** [reply]
Because I don't have a docker based environment to run it on.

**Comment by eliandoran** [comment]
@joellti , This method of installation was carried over from the original Trilium. In my opinion this installation method is overly complicated since it's more like installing a development environment. We should probably get rid of it.  

I would recommend downloading the server build which is available on the latest release and using that instead, if you are on x86_64.

**Reply by joellti** [reply]
unfortunately the server build I need is for arm64 which is not available. can we add that by any chance. or you can show me how to build it?

**Reply by TuxmAL** [reply]
> unfortunately the server build I need is for arm64 which is not available. can we add that by any chance. or you can show me how to build it?

Me too! I used it on a Raspberry PI 3+ and I have no chance to use docker and x86_64.

**Comment by Nriver** [comment]
I wrote an tutorial to build and run Trilium on android which is arm64. https://github.com/zadam/trilium/discussions/4542

It is the hard way.

**Reply by joellti** [reply]
i am aware of this procedure. i assume it applies to original trilium based on the publish date. but since i was able to manually install the original trilium by following the procedure i mentioned in my first post i didn't attempt this one. if this procedure was tested for trilium-next i am willing to give it a try.

**Reply by TuxmAL** [reply]
I always used a similar procedure  installing the original Trilium on my Raspberry PI with no problems. With TriliumNext v0.90.12, I   have problems on compiling lzma-native:
```
 error code 1
8187 error path /var/www/nodejs/trilium/node_modules/lzma-native
```

**Reply by joellti** [reply]
> I always used a similar procedure installing the original Trilium on my Raspberry PI with no problems. With TriliumNext v0.90.12, I have problems on compiling lzma-native:
> 
> ```
>  error code 1
> 8187 error path /var/www/nodejs/trilium/node_modules/lzma-native
> ```

that was what i ran into as well. Hopefully the procedure is updated for trilium-next. until then i am stuck at original trilium

**Reply by eliandoran** [reply]
@TuxmAL , @joellti , can you please provide more detailed logs? I would try it on my side but I do not have any Raspberry Pi devices to test on.

Please note that `npm i` issues have nothing to do with the TriliumNext application. You're seeing this issue on our repo and not on the original Trilium Notes because we are using newer dependencies.

**Reply by TuxmAL** [reply]
I'll give you more details tomorrow, but I issued an `npm install` after erasing `node_ modules`  into the  directory where I unpacked the TriliumNext tarball.
Previously, the same operation doesn't issued any error.

**Reply by TuxmAL** [reply]
[2024-12-20T16_02_37_663Z-debug-0.log](https://github.com/user-attachments/files/18213779/2024-12-20T16_02_37_663Z-debug-0.log)
@eliandoran this is what I've done:
1. unpacked the new server x86_64 TrilliumNext tarball;
2. removed the `node_modules` directory;
3. removed the `package-lock.json` (not needed because the result is the same with or without removing it);
4. executed `npm install`.

The result is what showed in the attached log file.

Hope this will help.

**Reply by TuxmAL** [reply]
Sorry to write with much delay. I resolved the issue related to my inability to build the node_module folder.
I found that I'm missing lzma and lzma-dev pacjage on my Raspberry Pi. Aster the installation I build successfully the folder for v0.90.4

**Comment by Nriver** [comment]
I've made it run on my phone. In short, the old way of using release files does not work now, we have to directly run from source code.

This one from the source code, it works.
[package src.json](https://github.com/user-attachments/files/18206978/package.src.json)

This one from release file will not work. Not sure which one cause the error.
[package release.json](https://github.com/user-attachments/files/18206982/package.release.json)

I'll write a detailed tutorial later.

![1.png](https://github.com/user-attachments/assets/ceb7a20e-4fdc-4dc3-92fc-b869ed689eb3)

![2.png](https://github.com/user-attachments/assets/08f034f3-644e-4866-8702-b5c36e83cc99)

**Reply by Nriver** [reply]
I write a new tutorial here https://github.com/orgs/TriliumNext/discussions/827

**Reply by joellti** [reply]
I just tried the new tutorial and it worked great for me. Thank you @Nriver for taking the efforts to make it work again. Since I use Debian as my main distro I modified the procedure slightly to adapt to that. If anyone needs that I will be happy to share the details.
